pkg/composableschemadsl/compiler/importer.go

```diff
@@ -26,25 +28,50 @@ func importFile(importContext importContext) (*CompiledSchema, error) {
 
 	newSourceFolder := filepath.Dir(filePath)
 
-	var schemaBytes []byte
+	currentLocallyVisitedFiles := importContext.locallyVisitedFiles.Copy()
+
+	if ok := currentLocallyVisitedFiles.Add(filePath); !ok {
+		return nil, fmt.Errorf("circular import detected: %s has been visited on this branch", filePath)
```

Comment:
this should be a well-typed developer error, attached to the node as an error, rather than returning as an internal error